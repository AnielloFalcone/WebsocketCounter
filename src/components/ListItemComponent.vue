<template>
    <div class="list-item">
        <div class="labels" v-if="!order.number">
            <div class="order-number-container">{{order.numberLabel}}</div>
            <div class="order-id-container">{{order.idLabel}}</div>
            <div class="order-amount-container">{{order.amountLabel}}</div>
        </div>
        <div class="item-detail" v-if="!!order.number">
            <div class="order-number-container">{{order.number}}</div>
            <div class="order-id-container">{{order.id}}</div>
            <div class="order-amount-container">{{order.amount | currency}}</div>
        </div>
    </div>
</template>

<script>
    export default {
        name: "ListItemComponent",
        props: ['order']
    }
</script>

<style lang="scss" scoped>
    .list-item {
        .item-detail, .labels {
            display: grid;
            grid-template-columns: 10% 65% 20%;
            width: 100%;
            height: 40px;
            align-items: center;
        }

        .order {
            &-id-container {
                white-space: nowrap;
                overflow: hidden;
                text-overflow: ellipsis;
            }

            &-amount-container {
                display: flex;
                justify-content: center;
            }
        }
    }
</style>
