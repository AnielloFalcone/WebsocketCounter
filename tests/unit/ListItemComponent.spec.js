import Vue from "vue";
import { shallowMount } from '@vue/test-utils';
import ListItemComponent from '../../src/components/ListItemComponent';

describe('ListItemComponent.vue', () => {
    let wrapper;
    const order = {
        id: '1673c3a3-8777-4064-bb96-43f02c514378',
        number: 12,
        amount: 123
    };

    beforeAll(() => {
        Vue.filter('currency', function (value) {
            return parseFloat(value).toFixed(2) + 'â‚¬';
        });
    });

    beforeEach(() => {
        wrapper = shallowMount(ListItemComponent, {
            propsData: {
                order
            },
        });
    });

    afterEach(() => {
        wrapper.destroy();
    });

    it('renders ListItemComponent when passed', () => {
        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.element).toBeDefined();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('order should be defined', () => {
        expect(wrapper.props().order).toBeDefined();
        expect(wrapper.props().order).toBeInstanceOf(Object);
    });

    it('should render table labels if order.number is undefined', () => {
        wrapper.props().order = {
            numberLabel: '',
            idLabel: 'Order id',
            amountLabel: 'Amount'
        };

        expect(wrapper.element).toThrowErrorMatchingSnapshot();
    });
});
