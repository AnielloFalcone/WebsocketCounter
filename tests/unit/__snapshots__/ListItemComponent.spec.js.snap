// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemComponent.vue renders ListItemComponent when passed 1`] = `
<div
  class="list-item"
>
  <!---->
   
  <div
    class="item-detail"
  >
    <div
      class="order-number-container"
    >
      12
    </div>
     
    <div
      class="order-id-container"
    >
      1673c3a3-8777-4064-bb96-43f02c514378
    </div>
     
    <div
      class="order-amount-container"
    >
      123.00â‚¬
    </div>
  </div>
</div>
`;

exports[`ListItemComponent.vue should render table labels if order.number is undefined 1`] = `"received is not a function"`;
